FerramAerospaceResearch/FARAeroComponents/VehicleOcclusion.cs

```diff
@@ -0,0 +1,828 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using FerramAerospaceResearch.UnityJobs;
+using Unity.Collections;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.Jobs;
+using UnityEngine.Profiling;
+using FerramAerospaceResearch.Settings;
+
+/* Theory of Job Control:
+ * At beginning of FixedUpdate cycle, pre-calculate the occlusion orientations we need.
+ * Convection occlusion calcuation and sun occlusion calculation are separate orientations.
+ * Weighted-average / lerp the 3 closest orientations; this can be 6 raycasts on first execution if we missed all caches.
+ * On exemplar system (4-core I5-6500 3.2GHz 16GB RAM):
+ *   <2ms at FixedUpdate.ObscenelyEarly to gather the raycast data
+ *  Expensive raycasts can cost ~4ms (10,000 rays), cheaper ones 0.5ms
+ *
+ * */
+namespace FerramAerospaceResearch.FARAeroComponents
+{
+    public class VehicleOcclusion : MonoBehaviour
+    {
+        public enum State { Invalid, Initialized, Running, Completed }
+        public State state = State.Invalid;
+        public static bool PassStarted { get; private set; } = false;
+        public static int JobsInCurrentPass { get; private set; } = 0;
+        private bool resetCoroutineRunning = false;
+        private IEnumerator resetWaitCoroutine;
```

Comment:
`resetCoroutineRunning` can be implemeted as a `resetCoroutineRunning is not null`

Reply:
Implemented in 0c9479b